src/components/Settings/ProfileForm.jsx

```diff
@@ -0,0 +1,224 @@
+import { useState } from "react";
+import Image from "next/image";
+import { useMutation } from "../../lib/apollo";
+import { UPDATE_SETTINGS } from "../../lib/graphql";
+
+const ProfileForm = ({ user }) => {
+  const [loading, setLoading] = useState(false);
+  const [newsletter, setNewsletter] = useState(user.newsletter || false);
+  const [journeyUpdates, setJourneyUpdates] = useState(
+    user.journeyUpdates || false
+  );
+  const [walletAddress, setWalletAddress] = useState(user.connectedWalletAddress || "");
+  const [username, setUsername] = useState(user.username || "");
+
+  const {
+    load: updateSettings,
+    error,
```

Comment:
use mutation also return "loading" state variable, you can use that instead of creating one in the component.

Reply:
I wasn't sure how to turn loading off once the query completes. Which meant the save button stayed disabled after query complete.